
<!DOCTYPE html>
<meta charset="utf-8">
<html lang="en">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>Teaser - Rapid Personalized Benchmarks for NGS Read Mapping</title>
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/bootstrap/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/DataTables-1.10.8/media/css/jquery.dataTables.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/c3/c3.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/extensions/bootstrap-select.min.css">
    <style>
body {
  padding-top: 60px;
  position: relative;
}

.sub-header {
  padding-bottom: 10px;
  border-bottom: 1px solid #eee;
}


.sidebar {
  display: none;
}

@media (min-width: 768px) {
  .sidebar {
    position: fixed;
    top: 51px;
    bottom: 0;
    left: 0;
    z-index: 1000;
    display: block;
    padding: 20px;
    overflow-x: hidden;
    overflow-y: auto; /* Scrollable contents if viewport is shorter than content. */
    background-color: #f5f5f5;
    border-right: 1px solid #eee;
  }
}

.nav-sidebar {
  margin-right: -21px;
  margin-bottom: 20px;
  margin-left: -20px;
}
.nav-sidebar > li > a {
  padding-right: 20px;
  padding-left: 20px;
}

.nav-sidebar > .active > a,
.nav-sidebar > .active > a:hover,
.nav-sidebar > .active > a:focus {
  color: #fff;
  background-color: #428bca;
}

a.separator
{
color:#777777;
}

a.separator:hover
{
color:#777777;
}

.main {
  padding: 20px;
}
@media (min-width: 768px) {
  .main {
    padding-right: 40px;
    padding-left: 40px;
  }
}
.main .page-header {
  margin-top: 0;
}

.placeholders {
  margin-bottom: 30px;
  text-align: center;
}
.placeholders h4 {
  margin-bottom: 0;
}
.placeholder {
  margin-bottom: 20px;
}
.placeholder img {
  display: inline-block;
  border-radius: 50%;
}

section {
    padding-top: 60px;
    margin-top: -60px;
}

.bar {
  fill: steelblue;
}

.bar:hover {
  fill: brown;
}

.axis {
  font: 10px sans-serif;
}

.axis path,
.axis line {
  fill: none;
  stroke: #000;
  shape-rendering: crispEdges;
}

.x.axis path {
  display: none;
}

.main-container
{
   margin: 0 auto;
   max-width: 1300px;
}


    </style>
  </head>

<body data-spy="scroll" data-target="#sidebar">
    <nav class="navbar navbar-inverse navbar-fixed-top">
      <div class="container-fluid">
        <div class="navbar-header">
          <a class="navbar-brand" href="/">Teaser</a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
           <li><a href="start.html">Benchmark</a></li><li><a href="#" class="separator"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></a></li><li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">H1<span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li><a href="D1_n.html">D1</a></li><li><a href="D2_n.html">D2</a></li><li><a href="D3_n.html">D3</a></li><li><a href="D4_n.html">D4</a></li><li><a href="H1_n.html">H1</a></li><li><a href="H2_n.html">H2</a></li><li><a href="M1_n.html">M1</a></li><li><a href="M2_n.html">M2</a></li><li><a href="M3_n.html">M3</a></li><li><a href="M4_n.html">M4</a></li></ul></li><li><a href="#" class="separator"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></a></li><li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">NGM<span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li><a href="H1_n_ngm.html">NGM</a></li><li><a href="H1_n_bwa.html">BWA</a></li><li><a href="H1_n_bwamem.html">BWA-MEM</a></li><li><a href="H1_n_bwasw.html">BWA-SW</a></li><li><a href="H1_n_bowtie2.html">Bowtie 2</a></li></ul></li>
          </ul>
          <div class="nav navbar-nav navbar-right" style="padding:5px;">
           
          </div>
        </div>
      </div>
    </nav>

    <div class="container-fluid main-container">
      <div class="row">
        
        <div class="col-sm-3 col-md-2 sidebar">
          <div id="sidebar">
          <ul class="nav nav-sidebar">
            <li class="active"><a href="#section1">Mapping Quality Thresholds</a></li><li><a href="#section2">Basic Statistics</a></li><li><a href="#section3">Read Failure Statistics</a></li><li><a href="#section4">Advanced Statistics</a></li><li><a href="#section5">Timing</a></li><li><a href="#section6">Additional Information</a></li><li><a href="#section7">Errors and Warnings</a></li><li><a href="#section8">Subprocess Log</a></li>
          </ul>
          
          </div>
        </div>
        <div class="col-sm-9  col-sm-offset-3 col-md-offset-2  col-md-10 main" id="content">
            <section id="section1"><div class="panel panel-default"><div class="panel-heading">Mapping Quality Thresholds</div><div class="panel-body"><p>The plot below shows the percentages of correctly and wrongly mapped reads for all mapping quality thresholds for this mapper. The values at threshold 0 therefore correspond to the unfiltered results.</p><div id="plot_mapq_tresh"></div></div></div></section><section id="section2"><div class="panel panel-default"><div class="panel-heading">Basic Statistics</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Correctly Mapped</th><td class="col-md-2">300189</td><td class="col-md-2">95.688%</td></tr><tr><th>Wrongly Mapped</th><td>13242</td><td>4.221%</td></tr><tr><th>Not Mapped</th><td>285</td><td>0.091%</td></tr><tr><th>Total</th><td>313716</td><td>100.0%</td></tbody></table></div></div></div></section><section id="section3"><div class="panel panel-default"><div class="panel-heading">Read Failure Statistics</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Not mapped</th><td class="col-md-2">285</td><td class="col-md-2">0.091%</td></tr><tr><th>Missing in mapper output</th><td>0</td><td>0.0%</td></tr><tr><th>Mapped to wrong chromosome</th><td>10215</td><td>3.256%</td></tr><tr><th>Mapped to wrong position</th><td>3027</td><td>0.965%</td></tr><tr><th>Mapped to wrong strand</th><td>0</td><td>0.0%</td></tbody></table></div></div></div></section><section id="section4"><div class="panel panel-default"><div class="panel-heading">Advanced Statistics</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th>Missing in comparison alignment</th><td>0</td></tr><tr><th class="col-md-6">Secondary Alignments</th><td class="col-md-4">0</td></tr><tr><th><abbr title="Harmonic Mean of Precision and Recall">F-Measure</abbr></th><td>0.977966</td></tr><tr><th><abbr title="Correctly Mapped / (Correctly Mapped + Wrongly Mapped)">Precision</abbr></th><td>0.957751</td></tr><tr><th><abbr title="Correctly Mapped / (Correctly Mapped + Unmapped)">Recall</abbr></th><td>0.999051</td></tbody></table></div></div></div></section><section id="section5"><div class="panel panel-default"><div class="panel-heading">Timing</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Raw Mapping Time</th><td class="col-md-4">20.228s</td></tr><tr><th>Effective Mapping Time</th><td>17.508s</td></tr><tr><th>Effective Init Time</th><td>2.720s</td></tr><tr><th>Effective Time Measure</th><td>Wall clock</td></tr><tr><th>Mapping Time (Wall)</th><td>20.228s</td></tr><tr><th>Mapping Time (CPU)</th><td>63.060s</td></tr><tr><th>Mapping Time (CPU User)</th><td>58.250s</td></tr><tr><th>Mapping Time (CPU System)</th><td>4.810s</td></tr><tr><th>Init Time (Wall)</th><td>2.720s</td></tr><tr><th>Init Time (CPU)</th><td>2.260s</td></tr><tr><th>Init Time (CPU User)</th><td>0.540s</td></tr><tr><th>Init Time (CPU System)</th><td>1.720s</td></tr></tbody></table></div></div></div></section><section id="section6"><div class="panel panel-default"><div class="panel-heading">Additional Information</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Mapper Memory Usage</th><td class="col-md-4">6794 MB</td></tr><tr><th>Total Test Runtime (Wall)</th><td>73.681s</td></tr><tr><th>Mapper Command Line:</th><td>&nbsp;</td></tr><tr><td colspan=2>/project/teaser/software/ngm/ngm --qry2 reads2.fastq --output out_ngm.sam --qry1 reads1.fastq --ref /scratch/mate/refs/GRCh37.fa -t 4  --no-progress</td></tr></tbody></table></div></div></div></section><section id="section7"><div class="panel panel-default"><div class="panel-heading">Errors and Warnings</div><div class="panel-body"><i>No problems were encountered.</i></div></div></section><section id="section8"><div class="panel panel-default"><div class="panel-heading">Subprocess Log</div><div class="panel-body"><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</h1></div><div class="panel-body"><div class="table-responsive"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>5448048640</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse121" class="collapsed">stderr</a></h4></div><div id="collapse121" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse122" class="collapsed">stdout</a></h4></div><div id="collapse122" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-09-11.19:51:49<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads_base.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/GRCh37.fa --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x1dfda000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/GRCh37.fa-enc.ngm<br>
[SEQPROV] Reading from disk took 9.39s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/GRCh37.fa-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 29.80s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 1<br>
[INPUT] Average read length: 60 (min: 60, max: 62)<br>
[INPUT] Corridor width: 14<br>
[INPUT] Not enough reads to estimate parameter<br>
[INPUT] Sensitivity parameter neither set nor estimated. Falling back to default.<br>
[INPUT] Initializing took 0.000s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 1<br>
[MAIN] Done (1 reads mapped (100.00%), 0 reads not mapped, 1 lines written)(elapsed: 41.493038s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>8.99</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>41.102383852</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>1.01</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/H1_n</small></td></tr></tbody></table></div></div></div><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</h1></div><div class="panel-body"><div class="table-responsive"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>5545762816</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse123" class="collapsed">stderr</a></h4></div><div id="collapse123" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse124" class="collapsed">stdout</a></h4></div><div id="collapse124" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-09-11.19:52:31<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads_base.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/GRCh37.fa --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x89639000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/GRCh37.fa-enc.ngm<br>
[SEQPROV] Reading from disk took 0.33s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/GRCh37.fa-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 1.14s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 1<br>
[INPUT] Average read length: 60 (min: 60, max: 62)<br>
[INPUT] Corridor width: 14<br>
[INPUT] Not enough reads to estimate parameter<br>
[INPUT] Sensitivity parameter neither set nor estimated. Falling back to default.<br>
[INPUT] Initializing took 0.000s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 1<br>
[MAIN] Done (1 reads mapped (100.00%), 0 reads not mapped, 1 lines written)(elapsed: 2.671800s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>1.72</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>2.71994805336</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>0.54</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/H1_n</small></td></tr></tbody></table></div></div></div><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --qry2 reads2.fastq --output out_ngm.sam --qry1 reads1.fastq --ref /scratch/mate/refs/GRCh37.fa -t 4  --no-progress</h1></div><div class="panel-body"><div class="table-responsive"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --qry2 reads2.fastq --output out_ngm.sam --qry1 reads1.fastq --ref /scratch/mate/refs/GRCh37.fa -t 4  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>6794600448</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse125" class="collapsed">stderr</a></h4></div><div id="collapse125" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse126" class="collapsed">stdout</a></h4></div><div id="collapse126" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-09-11.19:52:34<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 1 --parse_all 1 --pe_delimiter / --qry1 reads1.fastq --qry2 reads2.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/GRCh37.fa --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x64a68000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/GRCh37.fa-enc.ngm<br>
[SEQPROV] Reading from disk took 0.27s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/GRCh37.fa-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 1.44s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 156858<br>
[INPUT] Average read length: 100 (min: 100, max: 102)<br>
[INPUT] Corridor width: 20<br>
[INPUT] Average kmer hits pro read: 19.732021<br>
[INPUT] Max possible kmer hit: 29<br>
[INPUT] Estimated sensitivity: 0.680415<br>
[INPUT] Initializing took 0.613s<br>
[INPUT] Input is Fastq<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Valid pairs found: 96.02%<br>
[MAIN] Estimated insert size: 500 bp<br>
[MAIN] Alignments computed: 313545<br>
[MAIN] Done (313431 reads mapped (99.91%), 285 reads not mapped, 313716 lines written)(elapsed: 20.187645s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>4.81</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>20.2284250259</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>58.25</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/H1_n</small></td></tr></tbody></table></div></div></div></div></div></section><p></p><div style="margin:300px; height:300px;">&nbsp;</div>
        </div>
      </div>
    </div>


<script src="http://teaser.cibiv.univie.ac.at/static/js/d3.v3.min.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/c3/c3.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/js/jquery.js"></script>
<script>

var cols=[["Correct", 0.9568813831618407, 0.9351770391054329, 0.9344948934705275, 0.921317369850438, 0.9185824121179665, 0.9114963852656542, 0.9067149906284665, 0.9042860421527751, 0.9008657511889735, 0.8981339810529269, 0.8958580372056255, 0.893384462379987, 0.8915037804893599, 0.8888485126675082, 0.8871527113695189, 0.8856003519106453, 0.8838662994555585, 0.8824573818358006, 0.8811536548980606, 0.8799136798888166, 0.8788936490328833, 0.8778544925984011, 0.8769874663708577, 0.8762925703502531, 0.8755849239439493, 0.8749123410983182, 0.8744055132667763, 0.8739082482245087, 0.8734970482857106, 0.8731049739254613, 0.8727702763008581, 0.8724642670440781, 0.872234760101493, 0.8720020655624833, 0.8718235601626949, 0.8716673679378801, 0.8715430516773133, 0.8714282982060207, 0.871348608295401, 0.8712593555955067, 0.8712051664562853, 0.8710999757742671, 0.8710457866350457, 0.8710011602850987, 0.8709660967244259, 0.8709278455673284, 0.8708800316209565, 0.8708290300781598, 0.8707812161317879, 0.8707206517997169, 0.8706919634318938, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], ["Wrong", -0.9568091275261719, -0.36344584741403446, -0.2969706625987439, -0.18316803642039314, -0.1442945044305819, -0.10708285206115291, -0.08779059733758487, -0.0739897709248452, -0.06459653828790195, -0.0578767641707041, -0.053974959844589214, -0.050362178061149517, -0.048122253355416894, -0.04581007301401548, -0.04450947157197719, -0.043281125765607685, -0.041908268687900596, -0.041330223602550245, -0.04082443415286868, -0.039957366524843156, -0.03916255453248642, -0.03858450944713607, -0.038439998175798476, -0.0381509756331233, -0.038006464361785716, -0.037645186183441745, -0.037283908005097774, -0.0369226298267538, -0.03685037419108501, -0.03670586291974742, -0.036489096012741036, -0.03620007347006586, -0.035911050927390685, -0.035766539656053094, -0.035549772749046714, -0.03540526147770913, -0.03526075020637154, -0.03518849457070274, -0.03518849457070274, -0.03497172766369636, -0.03497172766369636, -0.034754960756689976, -0.03461044948535239, -0.0344659382140148, -0.0344659382140148, -0.03410466003567083, -0.034032404400002034, -0.03396014876433324, -0.03374338185732686, -0.03367112622165807, -0.03367112622165807, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]];
var a=0;
var b=0;
var last_a=0;
var last_b=0;

function updateRange()
{
	if(last_a==a && last_b==b)
	{
		setTimeout(updateRange,500);
		return;
	}
 
        an=Math.floor(parseInt(a));
        bn=Math.floor(parseInt(b));
	var mn = 0;
	var mx = 0;
	for(i=an;i<bn;i++)
	{
		mx=Math.max(mx,cols[0][i+1]);
		mn=Math.min(mn,cols[1][i+1]);
	}
	r={}
	r.max=mx;
	r.min=mn;
	chart.axis.range(r);
	last_a=a;
	last_b=b;
	setTimeout(updateRange,500);
}
setTimeout(updateRange,500);

var chart = c3.generate({
bindto: '#plot_mapq_tresh',
data: {
  columns: [["Correct", 0.9568813831618407, 0.9351770391054329, 0.9344948934705275, 0.921317369850438, 0.9185824121179665, 0.9114963852656542, 0.9067149906284665, 0.9042860421527751, 0.9008657511889735, 0.8981339810529269, 0.8958580372056255, 0.893384462379987, 0.8915037804893599, 0.8888485126675082, 0.8871527113695189, 0.8856003519106453, 0.8838662994555585, 0.8824573818358006, 0.8811536548980606, 0.8799136798888166, 0.8788936490328833, 0.8778544925984011, 0.8769874663708577, 0.8762925703502531, 0.8755849239439493, 0.8749123410983182, 0.8744055132667763, 0.8739082482245087, 0.8734970482857106, 0.8731049739254613, 0.8727702763008581, 0.8724642670440781, 0.872234760101493, 0.8720020655624833, 0.8718235601626949, 0.8716673679378801, 0.8715430516773133, 0.8714282982060207, 0.871348608295401, 0.8712593555955067, 0.8712051664562853, 0.8710999757742671, 0.8710457866350457, 0.8710011602850987, 0.8709660967244259, 0.8709278455673284, 0.8708800316209565, 0.8708290300781598, 0.8707812161317879, 0.8707206517997169, 0.8706919634318938, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.870688775835469, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], ["Wrong", -0.9568091275261719, -0.36344584741403446, -0.2969706625987439, -0.18316803642039314, -0.1442945044305819, -0.10708285206115291, -0.08779059733758487, -0.0739897709248452, -0.06459653828790195, -0.0578767641707041, -0.053974959844589214, -0.050362178061149517, -0.048122253355416894, -0.04581007301401548, -0.04450947157197719, -0.043281125765607685, -0.041908268687900596, -0.041330223602550245, -0.04082443415286868, -0.039957366524843156, -0.03916255453248642, -0.03858450944713607, -0.038439998175798476, -0.0381509756331233, -0.038006464361785716, -0.037645186183441745, -0.037283908005097774, -0.0369226298267538, -0.03685037419108501, -0.03670586291974742, -0.036489096012741036, -0.03620007347006586, -0.035911050927390685, -0.035766539656053094, -0.035549772749046714, -0.03540526147770913, -0.03526075020637154, -0.03518849457070274, -0.03518849457070274, -0.03497172766369636, -0.03497172766369636, -0.034754960756689976, -0.03461044948535239, -0.0344659382140148, -0.0344659382140148, -0.03410466003567083, -0.034032404400002034, -0.03396014876433324, -0.03374338185732686, -0.03367112622165807, -0.03367112622165807, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, -0.033598870585989274, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]],
  type: "area"
},
grid: {
  y: {
    show: true
  }
},
axis: {
  x: {
    label: "Mapping Quality Threshold"
  },

  y: {
    label: { text: "Read Count", position: "outer-middle" },
    tick: {
       format: function(v) { if(v>=0){return Math.min(100,(Math.round(100000*v)/1000))+"%";}else{ return Math.min(100,(Math.round(100000*(-v/22.667749))/1000))+"%";} }
    }
  }
},
legend: {
	position: "right"
},
subchart: {
	show: true,
	onbrush: function (d) {
		a = d[0];
		b = d[1];
	}
}
});

</script>
<script src="http://teaser.cibiv.univie.ac.at/static/bootstrap/js/bootstrap.min.js"></script>
<script type="javascript">$('#content').scrollspy();</script>
<script src="http://teaser.cibiv.univie.ac.at/static/extensions/bootstrap-select.min.js"></script>

<script src="http://teaser.cibiv.univie.ac.at/static/DataTables-1.10.8/media/js/jquery.dataTables.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/extensions/dataTables.bootstrap.js"></script>
</body>
</html>
