
<!DOCTYPE html>
<meta charset="utf-8">
<html lang="en">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>Teaser - Rapid Personalized Benchmarks for NGS Read Mapping</title>
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/bootstrap/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/DataTables-1.10.8/media/css/jquery.dataTables.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/c3/c3.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/extensions/bootstrap-select.min.css">
    <link rel="stylesheet" href="http://maxcdn.bootstrapcdn.com/bootstrap/3.3.5/css/bootstrap.min.css">
    <style>
body {
  padding-top: 60px;
  position: relative;
}

.sub-header {
  padding-bottom: 10px;
  border-bottom: 1px solid #eee;
}


.sidebar {
  display: none;
}

@media (min-width: 768px) {
  .sidebar {
    position: fixed;
    top: 51px;
    bottom: 0;
    left: 0;
    z-index: 1000;
    display: block;
    padding: 20px;
    overflow-x: hidden;
    overflow-y: auto; /* Scrollable contents if viewport is shorter than content. */
    background-color: #f5f5f5;
    border-right: 1px solid #eee;
  }
}

.nav-sidebar {
  margin-right: -21px;
  margin-bottom: 20px;
  margin-left: -20px;
}
.nav-sidebar > li > a {
  padding-right: 20px;
  padding-left: 20px;
}

.nav-sidebar > .active > a,
.nav-sidebar > .active > a:hover,
.nav-sidebar > .active > a:focus {
  color: #fff;
  background-color: #428bca;
}

a.separator
{
color:#777777;
}

a.separator:hover
{
color:#777777;
}

.main {
  padding: 20px;
}
@media (min-width: 768px) {
  .main {
    padding-right: 40px;
    padding-left: 40px;
  }
}
.main .page-header {
  margin-top: 0;
}

.placeholders {
  margin-bottom: 30px;
  text-align: center;
}
.placeholders h4 {
  margin-bottom: 0;
}
.placeholder {
  margin-bottom: 20px;
}
.placeholder img {
  display: inline-block;
  border-radius: 50%;
}

section {
    padding-top: 60px;
    margin-top: -60px;
}

.bar {
  fill: steelblue;
}

.bar:hover {
  fill: brown;
}

.axis {
  font: 10px sans-serif;
}

.axis path,
.axis line {
  fill: none;
  stroke: #000;
  shape-rendering: crispEdges;
}

.x.axis path {
  display: none;
}

.main-container
{
   margin: 0 auto;
   max-width: 1300px;
}


    </style>
  </head>

<body data-spy="scroll" data-target="#sidebar">
    <nav class="navbar navbar-inverse navbar-fixed-top">
      <div class="container-fluid">
        <div class="navbar-header">
          <a class="navbar-brand" href="/">Teaser</a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
           <li><a href="start.html">Benchmark</a></li><li><a href="#" class="separator"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></a></li><li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">H2<span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li><a href="D1_n.html">D1</a></li><li><a href="D2_n.html">D2</a></li><li><a href="D3_n.html">D3</a></li><li><a href="D4_n.html">D4</a></li><li><a href="H1_n.html">H1</a></li><li><a href="H2_n.html">H2</a></li><li><a href="M1_n.html">M1</a></li><li><a href="M2_n.html">M2</a></li><li><a href="M3_n.html">M3</a></li><li><a href="M4_n.html">M4</a></li></ul></li><li><a href="#" class="separator"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></a></li><li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">NGM<span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li><a href="H2_n_ngm.html">NGM</a></li><li><a href="H2_n_bwa.html">BWA</a></li><li><a href="H2_n_bwamem.html">BWA-MEM</a></li><li><a href="H2_n_bwasw.html">BWA-SW</a></li><li><a href="H2_n_bowtie2.html">Bowtie 2</a></li></ul></li>
          </ul>
          <div class="nav navbar-nav navbar-right" style="padding:5px;">
           
          </div>
        </div>
      </div>
    </nav>

    <div class="container-fluid main-container">
      <div class="row">
        
        <div class="col-sm-3 col-md-2 sidebar">
          <div id="sidebar">
          <ul class="nav nav-sidebar">
            <li class="active"><a href="#section1">Mapping Quality Thresholds</a></li><li><a href="#section2">Basic Statistics</a></li><li><a href="#section3">Read Failure Statistics</a></li><li><a href="#section4">Advanced Statistics</a></li><li><a href="#section5">Timing</a></li><li><a href="#section6">Additional Information</a></li><li><a href="#section7">Errors and Warnings</a></li><li><a href="#section8">Subprocess Log</a></li>
          </ul>
          
          </div>
        </div>
        <div class="col-sm-9  col-sm-offset-3 col-md-offset-2  col-md-10 main" id="content">
            <section id="section1"><div class="panel panel-default"><div class="panel-heading">Mapping Quality Thresholds</div><div class="panel-body"><p>The plot below shows the percentages of correctly and wrongly mapped reads for all mapping quality thresholds for this mapper. The values at threshold 0 therefore correspond to the unfiltered results.</p><div id="plot_mapq_thresh"></div><div align="right" style="margin-top:5px;"><div class="dropdown">
  <button class="btn btn-default btn-xs dropdown-toggle" type="button" id="dropdownMenu1" data-toggle="dropdown" aria-haspopup="true" aria-expanded="true">
    <span class="glyphicon glyphicon-download-alt" aria-hidden="true"> Export
    <span class="caret"></span>
  </button>
  <ul class="dropdown-menu dropdown-menu-right" aria-labelledby="dropdownMenu1">
    <li><a href="javascript:exportSVG('plot_mapq_thresh');">Plot (.svg)</a></li>
    <li><a href="H2_n_ngm_mapqs.csv">Raw Data  (.csv)</a></li>
  </ul>
</div></div></div></div></section><section id="section2"><div class="panel panel-default"><div class="panel-heading">Basic Statistics</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Correctly Mapped</th><td class="col-md-2">298500</td><td class="col-md-2">95.15%</td></tr><tr><th>Wrongly Mapped</th><td>14972</td><td>4.772%</td></tr><tr><th>Not Mapped</th><td>242</td><td>0.077%</td></tr><tr><th>Total</th><td>313714</td><td>100.0%</td></tbody></table></div></div></div></section><section id="section3"><div class="panel panel-default"><div class="panel-heading">Read Failure Statistics</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Not mapped</th><td class="col-md-2">242</td><td class="col-md-2">0.077%</td></tr><tr><th>Missing in mapper output</th><td>0</td><td>0.0%</td></tr><tr><th>Mapped to wrong chromosome</th><td>10758</td><td>3.429%</td></tr><tr><th>Mapped to wrong position</th><td>4214</td><td>1.343%</td></tr><tr><th>Mapped to wrong strand</th><td>0</td><td>0.0%</td></tbody></table></div></div></div></section><section id="section4"><div class="panel panel-default"><div class="panel-heading">Advanced Statistics</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th>Missing in comparison alignment</th><td>2</td></tr><tr><th class="col-md-6">Secondary Alignments</th><td class="col-md-4">0</td></tr><tr><th><abbr title="Harmonic Mean of Precision and Recall">F-Measure</abbr></th><td>0.975100</td></tr><tr><th><abbr title="Correctly Mapped / (Correctly Mapped + Wrongly Mapped)">Precision</abbr></th><td>0.952200</td></tr><tr><th><abbr title="Correctly Mapped / (Correctly Mapped + Unmapped)">Recall</abbr></th><td>0.999200</td></tbody></table></div></div></div></section><section id="section5"><div class="panel panel-default"><div class="panel-heading">Timing</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Raw Mapping Time</th><td class="col-md-4">21.191s</td></tr><tr><th>Effective Mapping Time</th><td>15.922s</td></tr><tr><th>Effective Init Time</th><td>5.269s</td></tr><tr><th>Effective Time Measure</th><td>Wall clock</td></tr><tr><th>Mapping Time (Wall)</th><td>21.191s</td></tr><tr><th>Mapping Time (CPU)</th><td>66.910s</td></tr><tr><th>Mapping Time (CPU User)</th><td>62.170s</td></tr><tr><th>Mapping Time (CPU System)</th><td>4.740s</td></tr><tr><th>Init Time (Wall)</th><td>5.269s</td></tr><tr><th>Init Time (CPU)</th><td>4.610s</td></tr><tr><th>Init Time (CPU User)</th><td>0.540s</td></tr><tr><th>Init Time (CPU System)</th><td>4.070s</td></tr></tbody></table></div></div></div></section><section id="section6"><div class="panel panel-default"><div class="panel-heading">Additional Information</div><div class="panel-body"><div class="table-responsive"><table class="table table-striped"><tbody><tr><th class="col-md-6">Mapper Memory Usage</th><td class="col-md-4">6798 MB</td></tr><tr><th>Total Test Runtime (Wall)</th><td>55.550s</td></tr><tr><th>Mapper Command Line:</th><td>&nbsp;</td></tr><tr><td colspan=2>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads.fastq  --no-progress</td></tr></tbody></table></div></div></div></section><section id="section7"><div class="panel panel-default"><div class="panel-heading">Errors and Warnings</div><div class="panel-body"><i>No problems were encountered.</i></div></div></section><section id="section8"><div class="panel panel-default"><div class="panel-heading">Subprocess Log</div><div class="panel-body"><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</h1></div><div class="panel-body"><div class="table-responsive"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>5753180160</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse151" class="collapsed">stderr</a></h4></div><div id="collapse151" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse152" class="collapsed">stdout</a></h4></div><div id="collapse152" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-09-11.17:40:14<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads_base.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/GRCh37.fa --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x47158000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/GRCh37.fa-enc.ngm<br>
[SEQPROV] Reading from disk took 5.51s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/GRCh37.fa-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 7.59s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 1<br>
[INPUT] Average read length: 60 (min: 60, max: 62)<br>
[INPUT] Corridor width: 14<br>
[INPUT] Not enough reads to estimate parameter<br>
[INPUT] Sensitivity parameter neither set nor estimated. Falling back to default.<br>
[INPUT] Initializing took 0.000s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 1<br>
[MAIN] Done (1 reads mapped (100.00%), 0 reads not mapped, 1 lines written)(elapsed: 17.158684s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>11.86</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>17.3815619946</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>1.03</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/H2_n</small></td></tr></tbody></table></div></div></div><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</h1></div><div class="panel-body"><div class="table-responsive"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads_base.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>5496631296</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse153" class="collapsed">stderr</a></h4></div><div id="collapse153" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse154" class="collapsed">stdout</a></h4></div><div id="collapse154" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-09-11.17:40:32<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads_base.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/GRCh37.fa --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x83b92000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/GRCh37.fa-enc.ngm<br>
[SEQPROV] Reading from disk took 0.32s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/GRCh37.fa-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 3.42s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 1<br>
[INPUT] Average read length: 60 (min: 60, max: 62)<br>
[INPUT] Corridor width: 14<br>
[INPUT] Not enough reads to estimate parameter<br>
[INPUT] Sensitivity parameter neither set nor estimated. Falling back to default.<br>
[INPUT] Initializing took 0.000s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 1<br>
[MAIN] Done (1 reads mapped (100.00%), 0 reads not mapped, 1 lines written)(elapsed: 5.190912s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>4.07</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>5.26889109612</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>0.54</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/H2_n</small></td></tr></tbody></table></div></div></div><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads.fastq  --no-progress</h1></div><div class="panel-body"><div class="table-responsive"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/GRCh37.fa -t 4 --qry reads.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>6798966784</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse155" class="collapsed">stderr</a></h4></div><div id="collapse155" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse156" class="collapsed">stdout</a></h4></div><div id="collapse156" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-09-11.17:40:38<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/GRCh37.fa --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x31c98000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/GRCh37.fa-enc.ngm<br>
[SEQPROV] Reading from disk took 0.32s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/GRCh37.fa-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 1.25s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 313716<br>
[INPUT] Average read length: 100 (min: 100, max: 102)<br>
[INPUT] Corridor width: 20<br>
[INPUT] Average kmer hits pro read: 19.233557<br>
[INPUT] Max possible kmer hit: 29<br>
[INPUT] Estimated sensitivity: 0.663226<br>
[INPUT] Initializing took 1.172s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 313527<br>
[MAIN] Done (313473 reads mapped (99.92%), 243 reads not mapped, 313716 lines written)(elapsed: 21.147245s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>4.74</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>21.1909799576</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>62.17</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/H2_n</small></td></tr></tbody></table></div></div></div></div></div></section><p></p><div style="margin:300px; height:300px;">&nbsp;</div>
        </div>
      </div>
    </div>


<script src="http://teaser.cibiv.univie.ac.at/static/js/d3.v3.min.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/c3/c3.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/js/jquery.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/js/svgexport.js"></script>
<script>

var cols=[["Correct", 0.9515035988193068, 0.9363305431061413, 0.9357025826070879, 0.9232676896791345, 0.9206920953479921, 0.9134593929502668, 0.9079288778951529, 0.9052576550616166, 0.9014102016486354, 0.8981014554658064, 0.8954206697820307, 0.8922936177537503, 0.8897785881407907, 0.8866706618129889, 0.8844393300904646, 0.8824343191569392, 0.8806333156951873, 0.8788482503171678, 0.8773118190453726, 0.8758582658089853, 0.8746405962118362, 0.8735153675003346, 0.8725527072429028, 0.8717749287567658, 0.8710099007376145, 0.8702129965509987, 0.8697412292725221, 0.8692280229763415, 0.8687849442485831, 0.8683195522035995, 0.868013540995939, 0.8676915917045461, 0.8674174566643503, 0.8671241959236757, 0.8668819370509445, 0.8666811171959173, 0.8664962354246224, 0.8663751059882568, 0.8662762898691164, 0.8661774737499761, 0.8660659071638499, 0.8659925919786812, 0.8659352148772449, 0.8658618996920763, 0.8658140854408792, 0.8657216445552318, 0.8656483293700632, 0.8655973275021198, 0.8655750141848946, 0.865530387550444, 0.8655176370834582, 0.8655144494667117, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], ["Wrong", -0.951440050859725, -0.28907966813791675, -0.2301071616459434, -0.12252046607384116, -0.09189034955538086, -0.06475537081392331, -0.05090191562507612, -0.04187810536445089, -0.03596814512333719, -0.03221881550800699, -0.028469485892676786, -0.025736923630656466, -0.023830484843200432, -0.02236888177281747, -0.02135211441950759, -0.02065308686410704, -0.019890511349124628, -0.019064387874560346, -0.018428908278741666, -0.017856976642504855, -0.017666332763759254, -0.017475688885013652, -0.017030853167940577, -0.017030853167940577, -0.016840209289194972, -0.016395373572121896, -0.01607763377421256, -0.01582344193588509, -0.015569250097557616, -0.015505702137975749, -0.01544215417839388, -0.015378606218812013, -0.015315058259230145, -0.015251510299648276, -0.01512441438048454, -0.014997318461320805, -0.01487022254215707, -0.014743126622993333, -0.014743126622993333, -0.014743126622993333, -0.014679578663411465, -0.014679578663411465, -0.014616030703829597, -0.014488934784665862, -0.014488934784665862, -0.014488934784665862, -0.014425386825083994, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]];
var a=0;
var b=0;
var last_a=0;
var last_b=0;

function updateRange()
{
	if(last_a==a && last_b==b)
	{
		setTimeout(updateRange,500);
		return;
	}
 
        an=Math.floor(parseInt(a));
        bn=Math.floor(parseInt(b));
	var mn = 0;
	var mx = 0;
	for(i=an;i<bn;i++)
	{
		mx=Math.max(mx,cols[0][i+1]);
		mn=Math.min(mn,cols[1][i+1]);
	}
	r={}
	r.max=mx;
	r.min=mn;
	chart.axis.range(r);
	last_a=a;
	last_b=b;
	setTimeout(updateRange,500);
}
setTimeout(updateRange,500);

var chart = c3.generate({
bindto: '#plot_mapq_thresh',
data: {
  columns: [["Correct", 0.9515035988193068, 0.9363305431061413, 0.9357025826070879, 0.9232676896791345, 0.9206920953479921, 0.9134593929502668, 0.9079288778951529, 0.9052576550616166, 0.9014102016486354, 0.8981014554658064, 0.8954206697820307, 0.8922936177537503, 0.8897785881407907, 0.8866706618129889, 0.8844393300904646, 0.8824343191569392, 0.8806333156951873, 0.8788482503171678, 0.8773118190453726, 0.8758582658089853, 0.8746405962118362, 0.8735153675003346, 0.8725527072429028, 0.8717749287567658, 0.8710099007376145, 0.8702129965509987, 0.8697412292725221, 0.8692280229763415, 0.8687849442485831, 0.8683195522035995, 0.868013540995939, 0.8676915917045461, 0.8674174566643503, 0.8671241959236757, 0.8668819370509445, 0.8666811171959173, 0.8664962354246224, 0.8663751059882568, 0.8662762898691164, 0.8661774737499761, 0.8660659071638499, 0.8659925919786812, 0.8659352148772449, 0.8658618996920763, 0.8658140854408792, 0.8657216445552318, 0.8656483293700632, 0.8655973275021198, 0.8655750141848946, 0.865530387550444, 0.8655176370834582, 0.8655144494667117, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.8655112618499653, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], ["Wrong", -0.951440050859725, -0.28907966813791675, -0.2301071616459434, -0.12252046607384116, -0.09189034955538086, -0.06475537081392331, -0.05090191562507612, -0.04187810536445089, -0.03596814512333719, -0.03221881550800699, -0.028469485892676786, -0.025736923630656466, -0.023830484843200432, -0.02236888177281747, -0.02135211441950759, -0.02065308686410704, -0.019890511349124628, -0.019064387874560346, -0.018428908278741666, -0.017856976642504855, -0.017666332763759254, -0.017475688885013652, -0.017030853167940577, -0.017030853167940577, -0.016840209289194972, -0.016395373572121896, -0.01607763377421256, -0.01582344193588509, -0.015569250097557616, -0.015505702137975749, -0.01544215417839388, -0.015378606218812013, -0.015315058259230145, -0.015251510299648276, -0.01512441438048454, -0.014997318461320805, -0.01487022254215707, -0.014743126622993333, -0.014743126622993333, -0.014743126622993333, -0.014679578663411465, -0.014679578663411465, -0.014616030703829597, -0.014488934784665862, -0.014488934784665862, -0.014488934784665862, -0.014425386825083994, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, -0.014361838865502128, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]],
  type: "area"
},
grid: {
  y: {
    show: true
  }
},
axis: {
  x: {
    label: "Mapping Quality Threshold"
  },

  y: {
    label: { text: "Read Count", position: "outer-middle" },
    tick: {
       format: function(v) { if(v>=0){return Math.min(100,(Math.round(100000*v)/1000))+"%";}else{ return Math.min(100,(Math.round(100000*(-v/19.935885))/1000))+"%";} }
    }
  }
},
legend: {
	position: "right"
},
subchart: {
	show: true,
	onbrush: function (d) {
		a = d[0];
		b = d[1];
	}
}
});

</script>
<script src="http://teaser.cibiv.univie.ac.at/static/bootstrap/js/bootstrap.min.js"></script>
<script type="javascript">$('#content').scrollspy();</script>
<script src="http://teaser.cibiv.univie.ac.at/static/extensions/bootstrap-select.min.js"></script>

<script src="http://teaser.cibiv.univie.ac.at/static/DataTables-1.10.8/media/js/jquery.dataTables.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/extensions/dataTables.bootstrap.js"></script>
</body>
</html>
