
<!DOCTYPE html>
<meta charset="utf-8">
<html lang="en">
  <head>
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="description" content="">
    <meta name="author" content="">

    <title>Teaser</title>
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/bootstrap/css/bootstrap.min.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/DataTables-1.10.8/media/css/jquery.dataTables.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/c3/c3.css">
    <link rel="stylesheet" href="http://teaser.cibiv.univie.ac.at/static/extensions/bootstrap-select.min.css">
    <style>
body {
  padding-top: 60px;
  position: relative;
}

.sub-header {
  padding-bottom: 10px;
  border-bottom: 1px solid #eee;
}


.sidebar {
  display: none;
}

@media (min-width: 768px) {
  .sidebar {
    position: fixed;
    top: 51px;
    bottom: 0;
    left: 0;
    z-index: 1000;
    display: block;
    padding: 20px;
    overflow-x: hidden;
    overflow-y: auto; /* Scrollable contents if viewport is shorter than content. */
    background-color: #f5f5f5;
    border-right: 1px solid #eee;
  }
}

.nav-sidebar {
  margin-right: -21px;
  margin-bottom: 20px;
  margin-left: -20px;
}
.nav-sidebar > li > a {
  padding-right: 20px;
  padding-left: 20px;
}

.nav-sidebar > .active > a,
.nav-sidebar > .active > a:hover,
.nav-sidebar > .active > a:focus {
  color: #fff;
  background-color: #428bca;
}

a.separator
{
color:#777777;
}

a.separator:hover
{
color:#777777;
}

.main {
  padding: 20px;
}
@media (min-width: 768px) {
  .main {
    padding-right: 40px;
    padding-left: 40px;
  }
}
.main .page-header {
  margin-top: 0;
}

.placeholders {
  margin-bottom: 30px;
  text-align: center;
}
.placeholders h4 {
  margin-bottom: 0;
}
.placeholder {
  margin-bottom: 20px;
}
.placeholder img {
  display: inline-block;
  border-radius: 50%;
}

section {
    padding-top: 60px;
    margin-top: -60px;
}

.bar {
  fill: steelblue;
}

.bar:hover {
  fill: brown;
}

.axis {
  font: 10px sans-serif;
}

.axis path,
.axis line {
  fill: none;
  stroke: #000;
  shape-rendering: crispEdges;
}

.x.axis path {
  display: none;
}

.main-container
{
   margin: 0 auto;
   max-width: 1300px;
}


    </style>
  </head>

<body data-spy="scroll" data-target="#sidebar">
    <nav class="navbar navbar-inverse navbar-fixed-top">
      <div class="container-fluid">
        <div class="navbar-header">
          <a class="navbar-brand" href="/">Teaser</a>
        </div>
        <div id="navbar" class="navbar-collapse collapse">
          <ul class="nav navbar-nav">
           <li><a href="start.html">Benchmark</a></li><li><a href="#" class="separator"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></a></li><li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">M2<span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li><a href="D1_n.html">D1</a></li><li><a href="D2_n.html">D2</a></li><li><a href="D3_n.html">D3</a></li><li><a href="D4_n.html">D4</a></li><li><a href="H1_n.html">H1</a></li><li><a href="H2_n.html">H2</a></li><li><a href="M1_n.html">M1</a></li><li><a href="M2_n.html">M2</a></li><li><a href="M3_n.html">M3</a></li><li><a href="M4_n.html">M4</a></li></ul></li><li><a href="#" class="separator"><span class="glyphicon glyphicon-chevron-right" aria-hidden="true"></span></a></li><li class="dropdown"><a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">NGM<span class="caret"></span></a><ul class="dropdown-menu" role="menu"><li><a href="M2_n_ngm.html">NGM</a></li><li><a href="M2_n_bwa.html">BWA</a></li><li><a href="M2_n_bwamem.html">BWA-MEM</a></li><li><a href="M2_n_bwasw.html">BWA-SW</a></li><li><a href="M2_n_bowtie2.html">Bowtie 2</a></li></ul></li>
          </ul>
          <div class="nav navbar-nav navbar-right" style="padding:5px;">
           
          </div>
        </div>
      </div>
    </nav>

    <div class="container-fluid main-container">
      <div class="row">
        
        <div class="col-sm-3 col-md-2 sidebar">
          <div id="sidebar">
          <ul class="nav nav-sidebar">
            <li class="active"><a href="#section1">Mapping Quality Thresholds</a></li><li><a href="#section2">Basic Statistics</a></li><li><a href="#section3">Read Failure Statistics</a></li><li><a href="#section4">Advanced Statistics</a></li><li><a href="#section5">Timing</a></li><li><a href="#section6">Additional Information</a></li><li><a href="#section7">Errors and Warnings</a></li><li><a href="#section8">Subprocess Log</a></li>
          </ul>
          </div>
        </div>
        <div class="col-sm-9  col-sm-offset-3 col-md-offset-2  col-md-10 main" id="content">
            <section id="section1"><div class="panel panel-default"><div class="panel-heading">Mapping Quality Thresholds</div><div class="panel-body"><p>The plot below shows the percentages of correctly and wrongly mapped reads for all mapping quality thresholds for this mapper. The values at threshold 0 therefore correspond to the unfiltered results.</p><div id="plot_mapq_tresh"></div></div></div></section><section id="section2"><div class="panel panel-default"><div class="panel-heading">Basic Statistics</div><div class="panel-body"><table class="table table-striped"><tbody><tr><th class="col-md-6">Correctly Mapped</th><td class="col-md-2">297469</td><td class="col-md-2">92.221%</td></tr><tr><th>Wrongly Mapped</th><td>25066</td><td>7.771%</td></tr><tr><th>Not Mapped</th><td>26</td><td>0.008%</td></tr><tr><th>Total</th><td>322561</td><td>100.0%</td></tbody></table></div></div></section><section id="section3"><div class="panel panel-default"><div class="panel-heading">Read Failure Statistics</div><div class="panel-body"><table class="table table-striped"><tbody><tr><th class="col-md-6">Not mapped</th><td class="col-md-2">26</td><td class="col-md-2">0.008%</td></tr><tr><th>Missing in mapper output</th><td>0</td><td>0.0%</td></tr><tr><th>Mapped to wrong chromosome</th><td>14062</td><td>4.359%</td></tr><tr><th>Mapped to wrong position</th><td>11004</td><td>3.411%</td></tr><tr><th>Mapped to wrong strand</th><td>0</td><td>0.0%</td></tbody></table></div></div></section><section id="section4"><div class="panel panel-default"><div class="panel-heading">Advanced Statistics</div><div class="panel-body"><table class="table table-striped"><tbody><tr><th>Missing in comparison alignment</th><td>0</td></tr><tr><th class="col-md-6">Secondary Alignments</th><td class="col-md-4">0</td></tr><tr><th><abbr title="Harmonic Mean of Precision and Recall">F-Measure</abbr></th><td>0.959531</td></tr><tr><th><abbr title="Correctly Mapped / (Correctly Mapped + Wrongly Mapped)">Precision</abbr></th><td>0.922284</td></tr><tr><th><abbr title="Correctly Mapped / (Correctly Mapped + Unmapped)">Recall</abbr></th><td>0.999913</td></tbody></table></div></div></section><section id="section5"><div class="panel panel-default"><div class="panel-heading">Timing</div><div class="panel-body"><table class="table table-striped"><tbody><tr><th class="col-md-6">Raw Mapping Time</th><td class="col-md-4">55.932s</td></tr><tr><th>Effective Mapping Time</th><td>49.666s</td></tr><tr><th>Effective Init Time</th><td>6.266s</td></tr><tr><th>Effective Time Measure</th><td>Wall clock</td></tr><tr><th>Mapping Time (Wall)</th><td>55.932s</td></tr><tr><th>Mapping Time (CPU)</th><td>200.350s</td></tr><tr><th>Mapping Time (CPU User)</th><td>190.950s</td></tr><tr><th>Mapping Time (CPU System)</th><td>9.400s</td></tr><tr><th>Init Time (Wall)</th><td>6.266s</td></tr><tr><th>Init Time (CPU)</th><td>5.440s</td></tr><tr><th>Init Time (CPU User)</th><td>0.460s</td></tr><tr><th>Init Time (CPU System)</th><td>4.980s</td></tr></tbody></table></div></div></section><section id="section6"><div class="panel panel-default"><div class="panel-heading">Additional Information</div><div class="panel-body"><table class="table table-striped"><tbody><tr><th class="col-md-6">Mapper Memory Usage</th><td class="col-md-4">7716 MB</td></tr><tr><th>Total Test Runtime (Wall)</th><td>104.808s</td></tr><tr><th>Mapper Command Line:</th><td>&nbsp;</td></tr><tr><td colspan=2>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads.fastq  --no-progress</td></tr></tbody></table></div></div></section><section id="section7"><div class="panel panel-default"><div class="panel-heading">Errors and Warnings</div><div class="panel-body"><i>No problems were encountered.</i></div></div></section><section id="section8"><div class="panel panel-default"><div class="panel-heading">Subprocess Log</div><div class="panel-body"><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads_base.fastq  --no-progress</h1></div><div class="panel-body"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads_base.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>6740881408</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse211" class="collapsed">stderr</a></h4></div><div id="collapse211" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse212" class="collapsed">stdout</a></h4></div><div id="collapse212" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-08-25.12:34:18<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads_base.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/m_musculus.fasta --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x9ba73000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/m_musculus.fasta-enc.ngm<br>
[SEQPROV] Reading from disk took 9.96s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/m_musculus.fasta-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 15.94s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 1<br>
[INPUT] Average read length: 60 (min: 60, max: 62)<br>
[INPUT] Corridor width: 14<br>
[INPUT] Not enough reads to estimate parameter<br>
[INPUT] Sensitivity parameter neither set nor estimated. Falling back to default.<br>
[INPUT] Initializing took 0.000s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 1<br>
[MAIN] Done (1 reads mapped (100.00%), 0 reads not mapped, 1 lines written)(elapsed: 28.221970s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>11.16</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>28.5129179955</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>1.14</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/M2_n</small></td></tr></tbody></table></div></div><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads_base.fastq  --no-progress</h1></div><div class="panel-body"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads_base.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>6599077888</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse213" class="collapsed">stderr</a></h4></div><div id="collapse213" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse214" class="collapsed">stdout</a></h4></div><div id="collapse214" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-08-25.12:34:47<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads_base.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/m_musculus.fasta --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0x64296000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/m_musculus.fasta-enc.ngm<br>
[SEQPROV] Reading from disk took 0.50s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/m_musculus.fasta-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 4.32s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 1<br>
[INPUT] Average read length: 60 (min: 60, max: 62)<br>
[INPUT] Corridor width: 14<br>
[INPUT] Not enough reads to estimate parameter<br>
[INPUT] Sensitivity parameter neither set nor estimated. Falling back to default.<br>
[INPUT] Initializing took 0.000s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 1<br>
[MAIN] Done (1 reads mapped (100.00%), 0 reads not mapped, 1 lines written)(elapsed: 6.229601s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>4.98</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>6.26630306244</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>0.46</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/M2_n</small></td></tr></tbody></table></div></div><div class="panel panel-default"><div class="panel-heading"><h1 class="panel-title">/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads.fastq  --no-progress</h1></div><div class="panel-body"><table class="table table-striped" style="margin-bottom;5px;"><tbody><tr><th class="col-md-2">command</th><td class="col-md-8"><small>/project/teaser/software/ngm/ngm --output out_ngm.sam --ref /scratch/mate/refs/m_musculus.fasta -t 4 --qry reads.fastq  --no-progress</small></td></tr><tr><th class="col-md-2">memory</th><td class="col-md-8"><small>7716102144</small></td></tr><tr><th class="col-md-2">return</th><td class="col-md-8"><small>0</small></td></tr><tr><th class="col-md-2">stderr</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse215" class="collapsed">stderr</a></h4></div><div id="collapse215" class="panel-collapse collapse"><div class="panel-body"><pre>None</pre></div></div></div></small></td></tr><tr><th class="col-md-2">stdout</th><td class="col-md-8"><small><div class="panel panel-default" id="panel1"><div class="panel-heading"><h4 class="panel-title"><a href="javascript:void(0);" data-toggle="collapse" data-target="#collapse216" class="collapsed">stdout</a></h4></div><div id="collapse216" class="panel-collapse collapse"><div class="panel-body"><pre>[MAIN] NextGenMap 0.4.13<br>
[MAIN] Startup : x64 (build Jun 11 2015 10:48:29)<br>
[MAIN] Starting time: 2015-08-25.12:34:53<br>
[CONFIG] Parameter:  --affine 0 --argos_min_score 0 --block_multiplier 2 --bs_cutoff 6 --bs_mapping 0 --cpu_threads 4 --dualstrand 1 --fast_pairing 0 --format 1 --gap_extend_penalty 5 --gap_read_penalty 20 --gap_ref_penalty 20 --hard_clip 0 --keep_tags 0 --kmer 13 --kmer_min 0 --kmer_skip 2 --match_bonus 10 --match_bonus_tc 4 --match_bonus_tt 4 --max_cmrs 2147483647 --max_equal 1 --max_insert_size 1000 --min_identity 0.650000 --min_insert_size 0 --min_mq 0 --min_residues 0.500000 --min_score 0.000000 --mismatch_penalty 15 --mode 0 --no_progress 1 --no_unal 0 --ocl_threads 1 --output out_ngm.sam --overwrite 1 --pair_score_cutoff 0.900000 --paired 0 --parse_all 1 --pe_delimiter / --qry reads.fastq --qry_count -1 --qry_start 0 --ref /scratch/mate/refs/m_musculus.fasta --ref_mode -1 --silent_clip 0 --skip_mate_check 0 --skip_save 0 --step_count 4 --strata 0 --topn 1 --update_check 0<br>
[STATS] Exporting stats at 0xeee38000 (Key 0x73297d00, ID 0xd801a)<br>
[NGM] Opening for output (SAM): out_ngm.sam<br>
[NGM] NGM Core initialization<br>
[SEQPROV] Init sequence provider.<br>
[SEQPROV] Reading encoded reference from /scratch/mate/refs/m_musculus.fasta-enc.ngm<br>
[SEQPROV] Reading from disk took 0.47s<br>
[PREPROCESS] Reading RefTable from /scratch/mate/refs/m_musculus.fasta-ht-13-2.ngm<br>
[PREPROCESS] Reading from disk took 1.07s<br>
[INPUT] Initializing ReadProvider<br>
[INPUT] Input is Fastq<br>
[INPUT] Estimating parameter from data<br>
[INPUT] Reads found in files: 322561<br>
[INPUT] Average read length: 150 (min: 150, max: 152)<br>
[INPUT] Corridor width: 27<br>
[INPUT] Average kmer hits pro read: 19.689442<br>
[INPUT] Max possible kmer hit: 46<br>
[INPUT] Estimated sensitivity: 0.428031<br>
[INPUT] Initializing took 1.610s<br>
[INPUT] Input is Fastq<br>
[MAIN] Core initialization complete<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
Setting of real/effective user Id to 0/0 failed<br>
FATAL: Module fglrx not found.<br>
Error! Fail to load fglrx kernel module! Maybe you can switch to root user to load kernel module directly<br>
[OPENCL] Available platforms: 1<br>
[OPENCL] AMD Accelerated Parallel Processing<br>
[OPENCL] Selecting OpenCl platform: AMD Accelerated Parallel Processing<br>
[OPENCL] Platform: OpenCL 1.2 AMD-APP (1214.3)<br>
[OPENCL] 1 CPU device found.<br>
[OPENCL] Device 0: Intel(R) Core(TM) i5-2500K CPU @ 3.30GHz (Driver: 1214.3 (sse2,avx))<br>
[OPENCL] 4 CPU cores available.<br>
[MAIN] Alignments computed: 322560<br>
[MAIN] Done (322535 reads mapped (99.99%), 26 reads not mapped, 322561 lines written)(elapsed: 55.893677s)<br>
</pre></div></div></div></small></td></tr><tr><th class="col-md-2">systime</th><td class="col-md-8"><small>9.4</small></td></tr><tr><th class="col-md-2">time</th><td class="col-md-8"><small>55.9324519634</small></td></tr><tr><th class="col-md-2">usrtime</th><td class="col-md-8"><small>190.95</small></td></tr><tr><th class="col-md-2">working_directory</th><td class="col-md-8"><small>/scratch/mate/tests_paper/M2_n</small></td></tr></tbody></table></div></div></div></div></section><p></p><div style="margin:300px; height:300px;">&nbsp;</div>
        </div>
      </div>
    </div>


<script src="http://teaser.cibiv.univie.ac.at/static/js/d3.v3.min.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/c3/c3.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/js/jquery.js"></script>
<script>

var cols=[["Correct", 0.9222100625928119, 0.7261603231636806, 0.724132799687501, 0.7100176400742805, 0.7003078487479888, 0.6930472065748804, 0.6869119329367158, 0.6815889087645438, 0.6766069053605365, 0.6721891363184018, 0.6680844863452184, 0.6641689478889264, 0.6607990426616981, 0.6575624455529342, 0.6545428616602751, 0.6516999885293014, 0.6490369263488146, 0.6467675881461181, 0.6446470590058935, 0.6425823332640958, 0.6405052067670921, 0.6384745831021109, 0.63683768341492, 0.6351232790076916, 0.6334863793205007, 0.6317781752908752, 0.6301908786245082, 0.6288236953630476, 0.6274534119127855, 0.6260118241200889, 0.6248151512427107, 0.6235936768549205, 0.6224745086975797, 0.6214328452602763, 0.620391181822973, 0.6193774200848832, 0.6185744711852952, 0.6177467207752952, 0.6169158701764937, 0.6162183276961567, 0.6155703882366436, 0.614975151986756, 0.6143582144152578, 0.6136885736341343, 0.6131150387058572, 0.6125353033999771, 0.6119338667724864, 0.6113603318442093, 0.6107030918182917, 0.6098691410306888, 0.608830577782187, 0.6080059275609885, 0.6078137158552955, 0.6077765135896777, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], ["Wrong", -0.9221732727869877, -0.18071152620799824, -0.12298832086997925, -0.028659258736976917, -0.015635667475244145, -0.009822878155035734, -0.006916483494931529, -0.005113783009550439, -0.0038629296115309064, -0.003163923300872933, -0.0024649169902149597, -0.0019866495145016093, -0.001692331067908778, -0.001434802427140051, -0.00114048398054722, -0.0009565349514267006, -0.0008829553397784929, -0.0007357961164820774, -0.0006622165048338697, -0.0006254266990097659, -0.000588636893185662, -0.0005518470873615581, -0.0005150572815374542, -0.0005150572815374542, -0.00044147766988924647, -0.00040468786406514266, -0.0003678980582410387, -0.00033110825241693484, -0.00033110825241693484, -0.00033110825241693484, -0.00033110825241693484, -0.000294318446592831, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.00022073883494462323, -0.00018394902912051935, -0.00018394902912051935, -0.00018394902912051935, -0.00018394902912051935, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.00011036941747231162, -7.357961164820774e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]];
var a=0;
var b=0;
var last_a=0;
var last_b=0;

function updateRange()
{
	if(last_a==a && last_b==b)
	{
		setTimeout(updateRange,500);
		return;
	}
 
        an=Math.floor(parseInt(a));
        bn=Math.floor(parseInt(b));
	var mn = 0;
	var mx = 0;
	for(i=an;i<bn;i++)
	{
		mx=Math.max(mx,cols[0][i+1]);
		mn=Math.min(mn,cols[1][i+1]);
	}
	r={}
	r.max=mx;
	r.min=mn;
	chart.axis.range(r);
	last_a=a;
	last_b=b;
	setTimeout(updateRange,500);
}
setTimeout(updateRange,500);

var chart = c3.generate({
bindto: '#plot_mapq_tresh',
data: {
  columns: [["Correct", 0.9222100625928119, 0.7261603231636806, 0.724132799687501, 0.7100176400742805, 0.7003078487479888, 0.6930472065748804, 0.6869119329367158, 0.6815889087645438, 0.6766069053605365, 0.6721891363184018, 0.6680844863452184, 0.6641689478889264, 0.6607990426616981, 0.6575624455529342, 0.6545428616602751, 0.6516999885293014, 0.6490369263488146, 0.6467675881461181, 0.6446470590058935, 0.6425823332640958, 0.6405052067670921, 0.6384745831021109, 0.63683768341492, 0.6351232790076916, 0.6334863793205007, 0.6317781752908752, 0.6301908786245082, 0.6288236953630476, 0.6274534119127855, 0.6260118241200889, 0.6248151512427107, 0.6235936768549205, 0.6224745086975797, 0.6214328452602763, 0.620391181822973, 0.6193774200848832, 0.6185744711852952, 0.6177467207752952, 0.6169158701764937, 0.6162183276961567, 0.6155703882366436, 0.614975151986756, 0.6143582144152578, 0.6136885736341343, 0.6131150387058572, 0.6125353033999771, 0.6119338667724864, 0.6113603318442093, 0.6107030918182917, 0.6098691410306888, 0.608830577782187, 0.6080059275609885, 0.6078137158552955, 0.6077765135896777, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.6077734134008761, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], ["Wrong", -0.9221732727869877, -0.18071152620799824, -0.12298832086997925, -0.028659258736976917, -0.015635667475244145, -0.009822878155035734, -0.006916483494931529, -0.005113783009550439, -0.0038629296115309064, -0.003163923300872933, -0.0024649169902149597, -0.0019866495145016093, -0.001692331067908778, -0.001434802427140051, -0.00114048398054722, -0.0009565349514267006, -0.0008829553397784929, -0.0007357961164820774, -0.0006622165048338697, -0.0006254266990097659, -0.000588636893185662, -0.0005518470873615581, -0.0005150572815374542, -0.0005150572815374542, -0.00044147766988924647, -0.00040468786406514266, -0.0003678980582410387, -0.00033110825241693484, -0.00033110825241693484, -0.00033110825241693484, -0.00033110825241693484, -0.000294318446592831, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.0002575286407687271, -0.00022073883494462323, -0.00018394902912051935, -0.00018394902912051935, -0.00018394902912051935, -0.00018394902912051935, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.0001471592232964155, -0.00011036941747231162, -7.357961164820774e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, -3.678980582410387e-05, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]],
  type: "area"
},
grid: {
  y: {
    show: true
  }
},
axis: {
  x: {
    label: "Mapping Quality Threshold"
  },

  y: {
    label: { text: "Read Count", position: "outer-middle" },
    tick: {
       format: function(v) { if(v>=0){return Math.min(100,(Math.round(100000*v)/1000))+"%";}else{ return Math.min(100,(Math.round(100000*(-v/11.866957))/1000))+"%";} }
    }
  }
},
legend: {
	position: "right"
},
subchart: {
	show: true,
	onbrush: function (d) {
		a = d[0];
		b = d[1];
	}
}
});

</script>
<script src="http://teaser.cibiv.univie.ac.at/static/bootstrap/js/bootstrap.min.js"></script>
<script type="javascript">$('#content').scrollspy();</script>
<script src="http://teaser.cibiv.univie.ac.at/static/extensions/bootstrap-select.min.js"></script>

<script src="http://teaser.cibiv.univie.ac.at/static/DataTables-1.10.8/media/js/jquery.dataTables.js"></script>
<script src="http://teaser.cibiv.univie.ac.at/static/extensions/dataTables.bootstrap.js"></script>
</body>
</html>
